Implement guardian form

msoucy opened the pull request:
Allows associating a guardian with students.

To add more than one, fill all previous slots, submit, and go back to the page. It's not pretty but it works.

msoucy commented:
Closes #36

msoucy committed:
Create form for tracking guardian data

bot190 committed:
fix: Make update_guardians handle a FieldList instead of a dict

update_guardians is passed a FieldList object which while it acts like a list of objects is not one. We can convert the individual forms to dictionaries though.

bot190 commented on signinapp/admin/users.py:
```diff
@@ -164,53 +170,65 @@ def edit_student_data():
-        nested.second_guardian_email.process_data(second_guardian.email)
+    for g in user.student_user_data.guardians:
+        nested.guardian.append_entry(g.user)
+    nested.guardian.append_entry()
```
This is a bit awkward since it always adds a guardian slot, but we can deal with that in the future.

msoucy replied:
That was intentional - the point was to always have space for one more. I do the same thing in the guardian list of students

bot190 committed:
fix: Admin new user page errored accessing non-existent form field

The role data was moved to the admin_data form but the User.make() call was not updated.

bot190 committed:
fix: Ensure user is a guardian when attempting to edit

This fixes a server error caused by passing a user id for someone without guardian data. While we shouldn't ever be generating a link like this, it's possible to do so manually.

This also changes the redirect on missing (or incorrect) user to go to the list of guardians instead of the user page.

bot190 committed:
fix: Make name and email validators optional in the guardian info form

The Optional validator stops the validator chain if the field is empty. This is useful on the Guardian Info form because we expect it to be blank in some places since we have room for two guardians but only want one to be required.

bot190 reviewed (commented):
I've pushed some fixes to various issues I found while testing as well.
